File: app/CausalityCSharp/CausalityLibrary/Type/Option.cs
using System;
using System.Collections.Generic;

namespace CausalityLibrary.Type
{
    [Serializable]
    public sealed class Option
    {
        public SerialNumber SerialNumber
        { get; }

        public string Description
        { get; }

        public readonly SerialNumber[] Actions;

        public bool IsMultiple
        { get; }

        public readonly List<SerialNumber> ChosenActions = new List<SerialNumber>();

        public Option(SerialNumber serialNumber, string description,
            SerialNumber[] actions, bool isMultiple)
        {
            SerialNumber = serialNumber;
            Description = description;
            Actions = actions;
            IsMultiple = isMultiple;
        }

        // Don't instantiate with default constructor!
        private Option() { }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Option.java
package com.inha.dsp.causality.type;

import java.io.Serializable;
import java.util.ArrayList;

public class Option implements Serializable {
    private SerialNumber serialNumber;
    private String description;
    private boolean multiple;
    public final SerialNumber[] Actions;
    public final ArrayList<SerialNumber> ChosenActions = new ArrayList<>();

    public SerialNumber getSerialNumber() {
        return serialNumber;
    }
    public String getDescription() {
        return description;
    }
    public boolean isMultiple() {
        return multiple;
    }

    public Option(SerialNumber sn, String desc, SerialNumber[] actions,
                  boolean multi)
    {
        serialNumber = sn;
        description = desc;
        Actions = actions;
        multiple = multi;
    }

    // Don't instantiate with default constructor!
    Option() {
        Actions = null;
    }
}
